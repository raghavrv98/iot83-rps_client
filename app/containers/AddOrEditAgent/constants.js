/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
