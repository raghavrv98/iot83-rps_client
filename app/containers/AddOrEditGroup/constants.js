/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
