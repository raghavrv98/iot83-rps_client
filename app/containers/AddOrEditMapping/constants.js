/*
 *
 * AddOrEditMapping constants
 *
 */

export const DEFAULT_ACTION = "app/AddOrEditMapping/DEFAULT_ACTION";

export const GET_PLANTS = "app/AddOrEditMapping/GET_PLANTS";
export const GET_PLANTS_SUCCESS = "app/AddOrEditMapping/GET_PLANTS_SUCCESS";
export const GET_PLANTS_FAILURE = "app/AddOrEditMapping/GET_PLANTS_FAILURE";

export const GET_EDIT_SAVE_MAP = "app/AddOrEditMapping/GET_EDIT_SAVE_MAP";
export const GET_EDIT_SAVE_MAP_SUCCESS = "app/AddOrEditMapping/GET_EDIT_SAVE_MAP_SUCCESS";
export const GET_EDIT_SAVE_MAP_FAILURE = "app/AddOrEditMapping/GET_EDIT_SAVE_MAP_FAILURE";

export const GET_MAPPING_DATA = "app/AddOrEditMapping/GET_MAPPING_DATA";
export const GET_MAPPING_DATA_SUCCESS = "app/AddOrEditMapping/GET_MAPPING_DATA_SUCCESS";
export const GET_MAPPING_DATA_FAILURE = "app/AddOrEditMapping/GET_MAPPING_DATA_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditMapping/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditMapping/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditMapping/GET_AGENT_DETAILS_FAILURE";