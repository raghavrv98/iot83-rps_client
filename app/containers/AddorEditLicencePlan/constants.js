/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";