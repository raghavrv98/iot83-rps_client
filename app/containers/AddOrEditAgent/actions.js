/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
