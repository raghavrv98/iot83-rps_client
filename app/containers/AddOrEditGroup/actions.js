/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
