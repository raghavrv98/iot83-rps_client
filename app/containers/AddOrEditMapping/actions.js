/*
 *
 * AddOrEditMapping actions
 *
 */

import { DEFAULT_ACTION, GET_PLANTS,GET_PIPELINES,GET_EDIT_SAVE_MAP,GET_MAPPING_DATA,GET_AGENT_DETAILS } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function getPlantList() {
  return {
    type: GET_PLANTS
  };
}

export function handelMapping(payload,expId,mapId) {
  return {
    type: GET_EDIT_SAVE_MAP,
    payload,
    expId,
    mapId
  };
}

export function getmappingDetails(agentId,mapId) {
  return {
    type: GET_MAPPING_DATA,
    agentId,mapId
  };
}

export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}