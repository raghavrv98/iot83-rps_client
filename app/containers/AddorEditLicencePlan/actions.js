/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}