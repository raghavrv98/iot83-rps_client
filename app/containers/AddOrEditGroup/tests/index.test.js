import React from 'react';
import { shallow } from 'enzyme';
import history from 'utils/history';
import { onSubmitHandler } from '../actions';
import { AddOrEditGroup, mapDispatchToProps } from '../index';
import { ON_SUBMIT_REQUEST, GET_GROUP_DETAILS } from '../constants';

const match = {
  params: {
    id: ""
  }
}

const submitSuccess = "this is submit success";
const submitError = "this is submit error";
const groupDetails = "this is group details success";
const groupDetailsError = "this is group details Error";

const simulateChangeInput = (wrapper, inputSelector, newValue, id) => {
  let input = wrapper.find(inputSelector)
  input.simulate('change', {
    target: { id, value: newValue },
  })
  return wrapper.find(inputSelector)
}

describe('<AddOrEditGroup />', () => {

  it('match snapshots', () => {
    const wrapper = shallow(<AddOrEditGroup match={match} />)
    expect(wrapper).toMatchSnapshot()
  })


  it("clicking back button", () => {
    let expectedCount = history.length + 1
    const wrapper = shallow(<AddOrEditGroup history={history} match={match} />);
    let backbutton = wrapper.find("span.cursor-pointer");
    expect(backbutton.length).toBe(1);
    backbutton.simulate('click');
    expect(history.length).toEqual(expectedCount);
    expect(history.location.pathname).toEqual("/manageGroups");
  });

  it('input change handling', () => {
    const wrapper = shallow(<AddOrEditGroup match={match} />);
    const nameInput = simulateChangeInput(wrapper, '[id="name"]', 'demoName', "name")
    const descriptionInput = simulateChangeInput(wrapper, '[id="description"]', 'demoDescription', 'description')
    expect(nameInput.props().value).toEqual('demoName')
    expect(descriptionInput.props().value).toEqual('demoDescription')
  })

  it('clicking submit button', () => {
    const wrapper = shallow(<AddOrEditGroup match={match} onSubmitHandler={onSubmitHandler} />);
    const onSubmitHandlerSpy = jest.spyOn(wrapper.instance(), "onSubmitHandler");
    wrapper.update();
    wrapper.instance().forceUpdate();
    let addOrEditForm = wrapper.find('form.contentForm')
    addOrEditForm.simulate('submit', { preventDefault: () => { } })
    expect(onSubmitHandlerSpy).toHaveBeenCalledTimes(1);
  })

  it('Input should call modalClose on click event', () => {
    let expectedCount = history.length + 1;
    const wrapper = shallow(<AddOrEditGroup history={history} match={match} />);
    wrapper.setState({ modalSuccess: "submitSuccess" })
    wrapper.instance().modalCloseHandler();
    expect(history.length + 1).toEqual(expectedCount);
    expect(history.location.pathname).toEqual("/manageGroups");
  })

  it("test componentDidMount", () => {
    spyOn(AddOrEditGroup.prototype, 'componentDidMount').and.callThrough();
    const wrapper = shallow(<AddOrEditGroup match={match} />);
    expect(AddOrEditGroup.prototype.componentDidMount).toHaveBeenCalledTimes(1);
  });

  it('unit testing on mapDispatchToProps', () => {
    const payload = "demo"
    const mockParams = {
      params: {
        id: "demoId"
      }
    }
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).onSubmitHandler(payload, mockParams.params.id);
    expect(dispatch.mock.calls[0][0]).
      toEqual({
        type: ON_SUBMIT_REQUEST,
        payload: "demo",
        id: "demoId"
      })

    mapDispatchToProps(dispatch).getGroupDetails(mockParams.params.id);
    expect(dispatch.mock.calls[1][0]).
      toEqual({
        type: GET_GROUP_DETAILS,
        id: "demoId"
      })

  });

  it("test component will recieve props", () => {
    spyOn(AddOrEditGroup.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<AddOrEditGroup history={history} match={match} submitSuccess="" submitError="" groupDetails="" groupDetailsError="" />);
    wrapper.setProps({ submitSuccess, submitError, groupDetails, groupDetailsError });
    expect(AddOrEditGroup.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  });

});
