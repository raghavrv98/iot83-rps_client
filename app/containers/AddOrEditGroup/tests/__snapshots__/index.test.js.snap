// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddOrEditGroup /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      AddOrEditGroup
    </title>
    <meta
      content="Description of AddOrEditGroup"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p
        className="pd-l-30"
      >
        <span
          className="cursor-pointer"
          onClick={[Function]}
        >
          <button
            className="btn btn-transparent"
          >
            <i
              className="far fa-long-arrow-left"
            />
          </button>
          <FormattedMessage
            defaultMessage="Manage Groups"
            id="app.containers.AddOrEditGroup.title"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </span>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="Create Group"
          id="app.containers.AddOrEditGroup.titleCreateGroup"
          values={Object {}}
        />
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right"
    />
  </div>
  <form
    className="contentForm"
    onSubmit={[Function]}
  >
    <h5
      className="formHeader"
    >
      <FormattedMessage
        defaultMessage="Create Group"
        id="app.containers.AddOrEditGroup.headingAdd"
        values={Object {}}
      >
        <Component />
      </FormattedMessage>
    </h5>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Group Name"
          id="app.containers.AddOrEditGroup.formName"
          values={Object {}}
        />
         :
        <sup>
          <i
            className="fas fa-asterisk"
          />
        </sup>
      </label>
      <input
        autoFocus={true}
        className="form-control"
        disabled={false}
        id="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Group Description"
          id="app.containers.AddOrEditGroup.formDescription"
          values={Object {}}
        />
         :
      </label>
      <textarea
        className="form-control"
        id="description"
        onChange={[Function]}
        rows="4"
        value=""
      />
    </div>
    <div
      className="form-group justify-content-end mt-3"
    >
      <button
        className="btn btn-danger"
        id="saveGroup"
      >
        <i
          className="far fa-check-circle"
        />
        <FormattedMessage
          defaultMessage="Save"
          id="app.common.messages.save"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </button>
    </div>
  </form>
</div>
`;
